src/common/librlist/rlist.c

```diff
@@ -1709,6 +1709,10 @@ json_t *rlist_to_R (struct rlist *rl)
     if (!rl)
         return NULL;
 
+    /*  Reset default sort to order nodes by "rank" */
+    zlistx_set_comparator (rl->nodes, by_rank);
+    zlistx_sort (rl->nodes);
+
```

Comment:
Remember that `zlistx_sort()` invalidates any cached list handles.  I didn't spot any but thought I'd remind just in case!

Reply:
Thanks! I double checked and librlist doesn't store any handles anywhere. When used it is only temporarily.